File: NirvanaSMS.Api.CSharp/Operations/MessageSender.cs
using NirvanaSMS.Api.Core.Base;
using NirvanaSMS.Api.Core.Entity;
using NirvanaSMS.Api.Core.Utility;

namespace NirvanaSMS.Api.Operations
{
    public class MessageSender : MessageSenderBase 
    {
        public override ProcessResult<string> SendMessage(ApiUser apiUser, MessageBase message)
        {
            OneToManyMessageFormatter oneToManyFormatter = new OneToManyMessageFormatter();
            ManyToManyMessageFormatter manyToManyFormatter = new ManyToManyMessageFormatter();

            MessageBase formattedMessage = null;
            if (message is OneToManyMessage)
                formattedMessage = oneToManyFormatter.Format(message);
            else if (message is ManyToManyMessage)
                formattedMessage = manyToManyFormatter.Format(message);

            return base.SendMessage(apiUser, formattedMessage);
        }
    }
}

File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/operations/MessageSender.java
package org.nirvanasms.api.operations;

import org.nirvanasms.api.core.base.MessageBase;
import org.nirvanasms.api.core.base.MessageSenderBase;
import org.nirvanasms.api.core.entity.ApiUser;
import org.nirvanasms.api.core.entity.ManyToManyMessage;
import org.nirvanasms.api.core.entity.OneToManyMessage;
import org.nirvanasms.api.core.utility.ManyToManyMessageFormatter;
import org.nirvanasms.api.core.utility.OneToManyMessageFormatter;
import org.nirvanasms.api.core.utility.ProcessResult;

public class MessageSender extends MessageSenderBase {
	@Override
	public ProcessResult<String> sendMessage(ApiUser apiUser, MessageBase message){
		OneToManyMessageFormatter oneToManyFormatter = new OneToManyMessageFormatter();
        ManyToManyMessageFormatter manyToManyFormatter = new ManyToManyMessageFormatter();

        MessageBase formattedMessage = null;
        if (message instanceof OneToManyMessage)
            formattedMessage = oneToManyFormatter.format(message);
        else if (message instanceof ManyToManyMessage)
            formattedMessage = manyToManyFormatter.format(message);
		return super.sendMessage(apiUser, formattedMessage);
		
	}
}
